TeamCode/src/main/java/org/firstinspires/ftc/teamcode/hardware/SensorCapstoneDetector.java

```diff
@@ -1,53 +1,56 @@
 package org.firstinspires.ftc.teamcode.hardware;
+
 import com.qualcomm.robotcore.hardware.DistanceSensor;
 import org.firstinspires.ftc.robotcore.external.Telemetry;
 import org.firstinspires.ftc.robotcore.external.navigation.DistanceUnit;
+import java.util.Arrays;
 
 public class SensorCapstoneDetector {
 
     private int location = 0;
-    private DistanceSensor middle;
+    private DistanceSensor left;
     private DistanceSensor right;
+    private double[] rightSensArray;
+    private double[] leftSensArray;
+    private int loopCycleNum;
 
-    public SensorCapstoneDetector(DistanceSensor middle, DistanceSensor right){
-        this.middle = middle;
-        this.right = right;
-    }
 
-    public int getLocation() {
-        return location;
+    public SensorCapstoneDetector(DistanceSensor left, DistanceSensor right){
+        this.left = left;
+        this.right = right;
+        rightSensArray = new double[300];
+        leftSensArray = new double[300];
+        loopCycleNum = 0;
     }
 
-    public void setLocation(int location) {
-        this.location = location;
+    public int getLoopCycleNum(){
+        return loopCycleNum;
     }
 
-    public void blueCapstoneDetection(){
-        if((right.getDistance(DistanceUnit.CM) >= 50) && (right.getDistance(DistanceUnit.CM) <= 70)){
-            setLocation(3);
-        }else if((middle.getDistance(DistanceUnit.CM) >= 50) && (middle.getDistance(DistanceUnit.CM) <= 70)){
-            setLocation(2);
-        }else{
-            setLocation(1);
+    public void finalLocation(double leftAverage, double rightAverage){
+        if(Math.abs((leftAverage-rightAverage)) > 200)
```

Comment:
**Set averages within this method**
double left_average = sum(left_array)/ len(left_array)
double right_average = sum(right_array)/ len(right_array)